### ISSUE TITLE
SuperLU's CMake install is misconfigured

### ISSUE THREAD

**Issue by alexchandel** [xiaoyeli/superlu#78]
SuperLU repeatedly uses `CMAKE_INSTALL_PREFIX` to assert its installation location during configuration, and even sets it.

```cmake
set(CMAKE_INSTALL_NAME_DIR "${CMAKE_INSTALL_PREFIX}/lib")
```

```cmake
set(CMAKE_INSTALL_RPATH "${CMAKE_INSTALL_PREFIX}/lib")
```

```cmake
if (NOT CMAKE_INSTALL_PREFIX)
  set(CMAKE_INSTALL_PREFIX /usr/local)
endif()
```

```cmake
install(FILES ${CMAKE_CURRENT_BINARY_DIR}/superlu.pc
	DESTINATION ${CMAKE_INSTALL_LIBDIR}/pkgconfig)
```

[This is one of the most grievous CMake sins you can commit](https://www.reddit.com/r/cpp/comments/6m7sp6/comment/dk00yz7/?utm_source=share&utm_medium=web2x&context=3).

You must not assume `CMAKE_INSTALL_PREFIX` is valid at configuration, nor set it.  You must pass relative paths to `install()`, because the user must be able to substitute an installation root at installation time.  Instead of passing your desired absolute path to `install()`, you should invoke installation as `cmake --install . --prefix /usr/local`.

**Comment by gruenich** [comment]
I think

    set(CMAKE_INSTALL_RPATH "${CMAKE_INSTALL_PREFIX}/lib")

and

    install(FILES ${CMAKE_CURRENT_BINARY_DIR}/superlu.pc
        DESTINATION ${CMAKE_INSTALL_LIBDIR}/pkgconfig)

are both ok, and can remain.

The other two issues mentioned, should be fixed. I give it a try, see https://github.com/xiaoyeli/superlu/pull/81.

**Comment by gruenich** [comment]
@alexchandel, can you please check current master if the issue persist or confirm that my commits fixed them?

**Comment by gruenich** [comment]
No response for a year, probably fixed by #81. @xiaoyeli, do you mind closing this?
